tests/testEnumUtils.cpp

```diff
@@ -0,0 +1,81 @@
+// Copyright (C) 2005 - 2023 Settlers Freaks (sf-team at siedler25.org)
+//
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "s25util/enumUtils.h"
+#include <boost/test/unit_test.hpp>
+#include <sstream>
+
+enum class InvalidBitset : int
+{
+};
+template<>
+struct IsBitset<InvalidBitset> : std::true_type
+{};
+
+enum class Bitset : unsigned
+{
+    None,
+    A = 1 << 0,
+    B = 1 << 1,
+    C = 1 << 2
+};
+MAKE_BITSET_STRONG(Bitset);
+
+// Check type traits
+static_assert(IsBitset<InvalidBitset>::value);
+static_assert(!IsValidBitset<InvalidBitset>);
+
+static_assert(IsBitset<Bitset>::value);
+static_assert(IsValidBitset<Bitset>);
+
+BOOST_AUTO_TEST_SUITE(EnumUtils)
+
+BOOST_AUTO_TEST_CASE(Operators)
+{
+    BOOST_REQUIRE(static_cast<unsigned>(Bitset{}) == 0);
+
+    {
+        Bitset b{};
+        b = b | Bitset::A;
+        BOOST_TEST(static_cast<unsigned>(b) == 0b001u);
+        b |= Bitset::B;
+        BOOST_TEST(static_cast<unsigned>(b) == 0b011u);
+	(b |= Bitset::A) = Bitset::C; // should compile
```

Comment:
You can even go one step further:

```suggestion
	(b |= Bitset::A) = Bitset::C;
	BOOST_TEST(b == Bitset::C);
```

Might need an extra set of parentheses or `BOOST_TEST_DONT_PRINT_LOG_VALUE(Bitset)` if it fails to compile when it can't print the enum class. Not sure

Reply:
I had those already. I used `BOOST_CHECK` to get around the missing `operator<<` instead of parentheses. Didn't think it added anything. :man_shrugging: But sure…

Reply:
It is just for clarity: You have an operation in a test and just having it compile is implicit. Asserting the expected behavior avoids later wondering what this is supposed to do. That's basically my main concern on review: Can you immediately see what this does in 5 years time?